platform/util/platform-util/src/main/java/org/codice/ddf/platform/util/WaitCondition.java

```diff
@@ -0,0 +1,133 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.platform.util;
+
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Class used to wait for a certain condition to be {@code true} or a timeout to expire. The
+ * condition is checked at regular interval. Both the timeout and polling interval can be
+ * configured using the {@link #within(long, TimeUnit)} and {@link #checkEvery(long, TimeUnit)}
+ * methods respectively.
+ * <p/>
+ * Usage examples (using Java 8 lambda expressions):
+ * <pre>
+ *     expect("Query response contains 5 elements").until(() -&gt; runQuery(), hasSize(5));
+ *     expect("Configuration object exists").
+ *         within(30, SECONDS).
+ *         checkEvery(1, SECONDS).
+ *         until(() -&gt; configAdmin.getConfiguration(pid, null) != null);
+ * </pre>
+ */
+public class WaitCondition {
```

Comment:
If you decide not to use Failsafe then we probably need to add unit tests for this class.